pkg/clustermesh/types/addressing.go

```diff
@@ -42,6 +44,42 @@ type AddrCluster struct {
 
 const AddrClusterLen = 20
 
+// MarshalJSON marshals the address as a string in the form
+// <addr>@<clusterID>, e.g. "1.2.3.4@1"
+func (a AddrCluster) MarshalJSON() ([]byte, error) {
+	var b bytes.Buffer
+	b.WriteByte('"')
+	if !a.addr.IsValid() {
+		b.WriteString("0.0.0.0")
```

Comment:
I'm not sure if it's a good idea to convert non valid address to a valid address, maybe just return error in that case instead?

Reply:
Hmm the issue here is that the JSON marshalling is used for inspecting the data from cilium-dbg and if we error out on e.g. an uninitialized AddrCluster then we cannot inspect that. It also may be an OK situation to have it uninitialized and non-valid if the address isn't really used for anything in some configuration.

But yeah, turning this into "0.0.0.0" isn't correct either. I'll fix this up so that even for invalid addresses we preserve `Unmarshal(Marshal(data)) == data`.